### PR TITLE
Don't drop user-input when a form including the widget has validation errors.

### PR DESCRIPTION
**Pull request by deiferni**
Currently when the widget is rendered with `select2js` it will drop user input in the add-form when other fields have validation errors. This PR fixes that issue by always marking user input as a `selected` option in the select field when rendering the form after it has been submitted once.

At the moment the choice field is initialized too late during the forms `update`. It won't be set during
the first call to `extract` which happens during super's update and should set the submitted value on the widget. Thus the value won't be saved on the widget correctly and won't be rendered as selected. This PR moves the choice-field initialization before the super call to make
sure that the choice_field is already available in the first call to `extract`.

The second call to `extract` will then happen in a form's `extractData` and correclty return the new value.

Fixes https://github.com/4teamwork/opengever.core/issues/2657.

### CODE REVIEW

File: ftw/keywordwidget/tests/tests.zcml [hunk 1]
```diff
@@ -0,0 +1,14 @@
+<configure
+    xmlns="http://namespaces.zope.org/zope"
+    xmlns:plone="http://namespaces.plone.org/plone"
+    i18n_domain="ftw.keywordwidget.tests">
+
+  <include package="Products.GenericSetup" file="meta.zcml" />
```
Comment (jone) [inline ftw/keywordwidget/tests/tests.zcml]: Why is this needed?

Comment (deiferni) [inline ftw/keywordwidget/tests/tests.zcml]: this is a copy-paste fragment, we don't do generic setup in here, removed ✂️

**Comment by deiferni** [comment]
also manually tested in `gever` and in an environment with `ftw.simplelayout`, with and withoud select2js. seems to work fine.

**Comment by deiferni** [comment]
(\cc @maethu)
